### PR TITLE
#issue-1026: Add Implementation for all 'java.time' classes

### PR DESCRIPTION
**Pull request by xavier-fernandez**
- Store ZoneDateTime data as an ISO-8601 string with timezone, with the
following format: “2015-09-30T17:20:29.393+02:00[Europe/Madrid]”, when
retrieved from the database the timestamp will be parsed.
- Specific implementation of zoned timestamps, like on the postgreSQL
’TIMESTAMP WITH TIMEZONE’ will be implemented in future commits.

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @xavier-fernandez,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.typesafe.com/contribute/cla](http://www.typesafe.com/contribute/cla)

File: slick/src/main/scala/slick/jdbc/SQLiteProfile.scala [hunk 1]
```diff
@@ -242,9 +244,28 @@ trait SQLiteProfile extends JdbcProfile {
...
      * date/time/timestamp literals. SQLite expects these values as milliseconds
      * since epoch. */
+    @deprecated(message = "java.util.Date is deprecated, use some time class of java.time package instead", since = "3.2.0")
     class DateJdbcType extends super.DateJdbcType {
       override def valueToSQLLiteral(value: Date) = value.getTime.toString
     }
+    class LocalDateJdbcType extends super.LocalDateJdbcType {
+      override def sqlType : Int = {
+        /**
+         * In SQLite, [[LocalDate]] is stored as a 'VARCHAR' in order to about some conversion issues.
```
Comment (cvogt) [inline slick/src/main/scala/slick/jdbc/SQLiteProfile.scala]: what are the conversion issues?

File: slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala [hunk 2]
```diff
@@ -301,6 +484,7 @@ trait JdbcTypesComponent extends RelationalTypesComponent { self: JdbcProfile =>
     implicit def byteArrayColumnType = columnTypes.byteArrayJdbcType
     implicit def charColumnType = columnTypes.charJdbcType
     implicit def clobColumnType = columnTypes.clobJdbcType
+    @deprecated(message = "java.util.Date is deprecated, use some time class of java.time package instead", since = "3.2.0")
```
Comment (cvogt) [inline slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala]: should this maybe be more specific and say use java.time.LocalDate instead?

File: slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala [hunk 3]
```diff
@@ -172,6 +186,175 @@ trait JdbcTypesComponent extends RelationalTypesComponent { self: JdbcProfile =>
...
+        r.updateString(idx, v.toString)
+      }
+      override def valueToSQLLiteral(value: Instant) = {
+        value match {
+          case null => "NULL"
+          case _ =>  "'" + value.toString + "'"
+        }
+      }
+    }
+
```
Comment (cvogt) [inline slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala]: there is a lot of copy and paste between the classes above, let's pull that together into a super class and not repeat

Comment (cvogt) [inline slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala]: I see that almost all cases encode dates as VARCHAR even on databases that do support time, etc. This seems very limited as you loose the ability to compare or operate on the dates on the server side. Shouldn't we rather implement mapping to native db time types where available?

**Comment by cvogt** [comment]
FYI: I think we'd prefer to squash these commits before merge. they seem easy enough to understand as a single commit

**Comment by xavier-fernandez** [comment]
@cvogt I will work on your comments tomorrow, thanks a lot for your feedback :)

**Comment by cvogt** [comment]
@xavier-fernandez you are welcome. In open source, we are all in this together :).

**Comment by xavier-fernandez** [comment]
@cvogt I am porting each class to use specific JBDC methods in order to use specific Date classes on each database, using the new data types.

JBDC will implement the java.time classes soon, so I will add a notice to all the classes from my code pointing the following URL: https://java.net/jira/browse/JPA_SPEC-63

Unfortunately, I am still having issues with SQLite, the generic implementation working on all slick databases, do not work on SQLite yet.

**Pull request closed without merging**